ddl/placement/rule.go

```diff
@@ -84,6 +84,10 @@ func NewRules(role PeerRoleType, replicas uint64, cnstr string) ([]*Rule, error)
 	constraints2 := map[string]int{}
 	err2 := yaml.UnmarshalStrict(cnstbytes, &constraints2)
 	if err2 == nil {
+		if replicas != 0 {
```

Comment:
I think `create placement policy p1 follower_constraints='{+region=r1: 1}'` may fail because we set the followerCount outside when it is 0:

https://github.com/pingcap/tidb/blob/e02089d5af6b4ec3d44d216196d59da18ef1fde0/ddl/placement/bundle.go#L90-L93

Reply:
OK, it makes sense.

Reply:
Addressed.